@here/harp-mapview/test/ThemeLoaderTest.ts

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// tslint:disable:only-arrow-functions
+//    Mocha discourages using arrow functions, see https://mochajs.org/#arrow-functions
+
+import { assert } from "chai";
+
+import { SolidLineStyle, StyleSet, Theme } from "@here/harp-datasource-protocol";
+import { getTestResourceUrl } from "@here/harp-test-utils";
+import { ThemeLoader } from "../lib/ThemeLoader";
+
+describe.only("ThemeLoader", function() {
+    describe("#expandReferences", function() {
+        it("supports $ref in technique attr values", async function() {
+            const theme: Theme = {
+                definitions: {
+                    roadColor: { type: "color", value: "#f00" }
+                },
+                styles: {
+                    tilezen: [
+                        {
+                            description: "roads",
+                            when: "kind == 'road",
+                            technique: "solid-line",
+                            attr: {
+                                lineColor: { $ref: "roadColor" }
+                            }
+                        }
+                    ]
+                }
+            };
+            const r = await ThemeLoader.expandReferences(theme);
+
+            const roadStyle = r.styles!.tilezen.find(s => s.description === "roads")!;
+            assert.exists(roadStyle);
+            assert.equal(roadStyle.technique, "solid-line");
+            const roadStyleCasted = (roadStyle as any) as SolidLineStyle;
+            assert.equal(roadStyleCasted.attr!.lineColor, "#f00");
+        });
+
+        it("supports $ref in Style", async function() {
+            const theme: Theme = {
+                definitions: {
+                    roadColor: { type: "color", value: "#f00" },
+                    roadTechnique: {
+                        description: "roads",
+                        when: "kind == 'road", // TODO: do we want `when` here ...
+                        technique: "solid-line",
+                        attr: {
+                            lineColor: { $ref: "roadColor" }
+                        }
+                    }
+                },
+                styles: {
+                    tilezen: [
+                        {
+                            when: "kind == 'road", // or here, or both with override ?
```

Comment:
@SahinHERE, @robertoraggi  Which place for `when` is better, here or in `definitions`.

I guess, that it should be here but with possibility to define base, default when in `definitions`

Reply:
Hey @zbigg,

I would leave out the 'when' condition in the 'roadStyle' defined in 'definitions'.
The idea would be to define styles and conditions in 'definitions' and use them potentially for different datasources. 

So in that way, we would be able to define a set of styles and conditions and could overwrite the given conditions depending on the data source. This would also allow to use the same styling for different data sources.

Let me give you an example to illustrate the idea better.
Also lets imagine that we would have another datasource where the kind attribute for a road is defined as 'roadClass' but is semantically the same as 'kind == road'

```js
 const baseTheme: Theme = {
                definitions: {
                    roadColor: { type: "color", value: "#f00" },
                    roadCondition: 'when: "kind == 'road"',
                },
                styles: {
                    tilezen: [
                        {
                            description: "roads",
                            when: { $ref : roadCondition }
                            technique: "solid-line",
                            attr: {
                                lineColor: { $ref: "roadColor" }
                            }
                        }
                    ]
                }
            };

 const otherDataSourceTheme: Theme = {
                extends: baseTheme,
                definitions: {
                    roadCondition: 'when: "kind == 'roadClass"',
                }
            };
```


One more thing to test would be to define a condition in 'definitions' and refer to the condition in a given style in a stlyset.

Reply:
@SahinHERE New patch contains support for
```
roadCondition: { type: "selector", value: "kind == 'road'"}
```